src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -104,14 +115,20 @@ def validate(self, data):
 
             snuba_filter = get_filter(conditions, params=params)
         except InvalidSearchQuery as err:
-            raise serializers.ValidationError({"conditions": f"Invalid conditions: {err}"})
+            data["discover_query_error"] = {"conditions": [f"Invalid conditions: {err}"]}
+            return data
 
         if orderby:
             snuba_filter.orderby = get_function_alias(orderby)
         try:
             resolve_field_list(fields, snuba_filter, resolved_equations=resolved_equations)
         except InvalidSearchQuery as err:
-            raise serializers.ValidationError({"fields": f"Invalid fields: {err}"})
+            # We don't know if the widget that this query belongs to is an
+            # Issue widget or Discover widget. Pass the error back to the
+            # Widget serializer to decide if whether or not to raise this
+            # error based on the Widget's type
+            data["discover_query_error"] = {"fields": f"Invalid fields: {err}"}
```

Comment:
Instead of raising a query error in `DashboardWidgetQuerySerializer` right away, we instead store it in `data` to bubble up to be used in `DashboardWidgetSerializer`. This is because we need to know if the widget that this query belongs to is an `issue` or `discover` widget when validating the query, since `issue` and `discover` queries have different validation rules. Information on the widget type is not available to us while we are validating in the `DashboardWidgetQuerySerializer`. We store both `issue` and `discover` validation errors in `data` if both or either exists.